### PR TITLE
DOC: Update SurfaceToolbox documentation to include "uniform remeshing"

### PR DESCRIPTION
**Pull request by jcfr**
Describe "uniform remeshing" tool added in https://github.com/Slicer/SlicerSurfaceToolbox/commit/94957ef2c87438e6d21f6071a036232eb8119717 and integrated in Slicer through https://github.com/Slicer/Slicer/commit/c0b92b4359c3d4a3b449307741e1d7a2be99aedb

### CODE REVIEW

File: Docs/user_guide/modules/surfacetoolbox.md [hunk 1]
```diff
@@ -31,6 +32,19 @@ These tools can be combined for multiple surface model processing effects.
...
 any cell) and treat degenerate cells.
 
+### Uniform remeshing
+
+Uniformly remesh the surface using [ACVD algorithm](https://github.com/pyvista/pyacvd).
+
+This resampling typically provides higher quality meshes than decimation, with similar computation time.
+
+- **Subdivide:** Number of subdivision to perform before remeshing. Each subdivision creates 4 triangles for each input triangle. This is needed if the required number of clusters are higher than the number of cells in the input mesh.
+- **Clusters:** Number of desired cells in the output mesh. Use higher number to preserve more details.
```
Comment (lassoan) [inline Docs/user_guide/modules/surfacetoolbox.md:42]: I've made a mistake when documenting this parameter (I'll fix that in a couple of minutes) in Surface toolbox. Clusters actually means the number of desired points (not cells).

```suggestion
- **Number of points:** Number of desired points in the output mesh. Use higher number to preserve more details.
```

File: Docs/user_guide/modules/surfacetoolbox.md [hunk 2]
```diff
@@ -31,6 +32,19 @@ These tools can be combined for multiple surface model processing effects.
 This tool can merge coincident points, remove unused points (i.e. not used by
 any cell) and treat degenerate cells.
 
+### Uniform remeshing
+
+Uniformly remesh the surface using [ACVD algorithm](https://github.com/pyvista/pyacvd).
+
+This resampling typically provides higher quality meshes than decimation, with similar computation time.
+
+- **Subdivide:** Number of subdivision to perform before remeshing. Each subdivision creates 4 triangles for each input triangle. This is needed if the required number of clusters are higher than the number of cells in the input mesh.
```
Comment (lassoan) [inline Docs/user_guide/modules/surfacetoolbox.md:41]:
```suggestion
- **Subdivide:** Number of subdivision to perform before remeshing. Each subdivision creates 4 triangles for each input triangle. This is needed if the required number of desired points is higher than the number of points in the input mesh, or there are some too large cells in the input mesh.
```

**Comment by jcfr** [comment]
Thanks for the review. Let's integrate once the SurfaceToolbox updates you will be working on is merged.

**Comment by lassoan** [comment]
I've pushed the fix for the SurfaceToolbox. The git hash update could be done at the same time as the doc update. I let you do it unless you say I should do it.

**Comment by jcfr** [comment]
> pushed the fix for the SurfaceToolbox [...] I will let you do it

:+1:  I am updating the hash has part of this pull request

**Comment by lassoan** [comment]
Thank you!

**Pull request merged**